rpcs3/Emu/system_utils.cpp

```diff
@@ -154,6 +156,94 @@ namespace rpcs3::utils
 		return rap_path;
 	}
 
+	std::string get_c00_unlock_edat_path(const std::string& title_id, const std::string& content_id)
+	{
+		const std::string home_dir = get_hdd0_dir() + "home";
+
+		std::string edat_path;
+
+		for (auto&& entry : fs::dir(home_dir))
+		{
+			if (entry.is_directory && check_user(entry.name))
+			{
+				edat_path = fmt::format("%s/%s/exdata/%s/%s.edat", home_dir, entry.name, title_id, content_id);
+				if (fs::is_file(edat_path))
+				{
+					return edat_path;
+				}
+			}
+		}
+
+		// Return a sample path tested for logging purposes
+		return edat_path;
+	}
+
+	bool verify_c00_unlock_edat(const std::string& title_id, const std::string& content_id)
+	{
+		const std::string edat_path = rpcs3::utils::get_c00_unlock_edat_path(title_id, content_id);
+		std::string rap_path        = rpcs3::utils::get_rap_file_path(content_id);
+		u128 k_licensee             = get_default_self_klic();
```

Comment:
These two variables should be moved to right before they are used

Reply:
What do you mean "moved to right"?

Reply:
You're basically initializing 2 variables here that might never be used if there is no edit.
This means you parse the home dir twice even if the first few checks fail.

So what I mean is: move these two lines down to where the variables are actually needed, in order to optimize this function

Reply:
Ah, okay.